### ISSUE TITLE
Extension throws error if prettier v3 plugins array is used

### ISSUE THREAD

**Issue by sagargurtu** [microsoft/vscode-eslint#1724]
Prettier Config:
```
module.exports = {
  printWidth: 80,
  semi: true,
  importOrderSeparation: true,
  importOrderSortSpecifiers: true,
  plugins: ["@trivago/prettier-plugin-sort-imports"],
};
```

Eslint Output:
```
[Error - 9:55:48 AM] An unexpected error occurred:
[Error - 9:55:48 AM] Error: Cannot find package '@trivago/prettier-plugin-sort-imports' imported from <redacted>/noop.js
    at new NodeError (file:///<redacted>/node_modules/prettier/index.mjs:18723:5)
    at packageResolve (file:///<redacted>/node_modules/prettier/index.mjs:19668:9)
    at moduleResolve (file:///<redacted>/node_modules/prettier/index.mjs:19700:20)
```

One of the solutions is to use `require`, i.e.
```
module.exports = {
  printWidth: 80,
  semi: true,
  importOrderSeparation: true,
  importOrderSortSpecifiers: true,
  plugins: [require("@trivago/prettier-plugin-sort-imports")],
};
```

But then this breaks prettier extension instead.

Eslint extension version: 2.4.2
Prettier extension version: 10.1.0
VSCode version: 1.83.1
Prettier version: 3.0.1
Eslint version: 8.46.0

**Comment by dbaeumer** [comment]
Does the validation happen correctly when you use eslint in the terminal. If this is the case can you please provide me with a GitHub repository I can clone that demos what you are experiencing.

**Comment by sagargurtu** [comment]
Yes, the validation happens correctly when eslint is used in the terminal.

I tried a few different workarounds and I was able to make the eslint extension error go away by using `require.resolve` in prettier's plugins array.

```
module.exports = {
  printWidth: 80,
  semi: true,
  importOrderSeparation: true,
  importOrderSortSpecifiers: true,
  plugins: [require.resolve("@trivago/prettier-plugin-sort-imports")],
};
```

However, I think the extension might need to be updated to be compatible with prettier v3's plugins array (if not already).

**Comment by felixgeissler** [comment]
> Yes, the validation happens correctly when eslint is used in the terminal.
> 
> I tried a few different workarounds and I was able to make the eslint extension error go away by using `require.resolve` in prettier's plugins array.
> 
> ```
> module.exports = {
>   printWidth: 80,
>   semi: true,
>   importOrderSeparation: true,
>   importOrderSortSpecifiers: true,
>   plugins: [require.resolve("@trivago/prettier-plugin-sort-imports")],
> };
> ```
> 
> However, I think the extension might need to be updated to be compatible with prettier v3's plugins array (if not already).

Thanks @sagargurtu for pointing me to the right direction!
I had to tweak my _prettier.config.cjs_ as well, to get ESLint in VSCode back working:
```
/* eslint-env es6 */

/** @type {import("prettier").Config} */
const config = {
  // ...
  // plugins: ['prettier-plugin-tailwindcss'],
  plugins: [require.resolve('prettier-plugin-tailwindcss')],
};

module.exports = config;
```
